sonar-db/src/main/java/org/sonar/db/component/ComponentUpdateDto.java

```diff
@@ -0,0 +1,179 @@
+import javax.annotation.Nullable;
+
+public class ComponentUpdateDto {
+  private String uuid;
+
+  /**
+   * if true, the component is being updated
+   * See https://jira.sonarsource.com/browse/SONAR-7700
+   */
+  private boolean bChanged;
+  private String bCopyComponentUuid;
+  private String bDescription;
+  private boolean bEnabled;
+  private String bLanguage;
+  private String bLongName;
+  private String bModuleUuid;
+  private String bModuleUuidPath;
+  private String bName;
+  private String bPath;
+  private String bQualifier;
+
+  public ComponentUpdateDto setUuid(String uuid) {
+    this.uuid = uuid;
+    return this;
+  }
+
+  public String getUuid() {
+    return uuid;
+  }
+
+  public boolean isBChanged() {
+    return bChanged;
+  }
+
+  @CheckForNull
+  public String getBCopyComponentUuid() {
+    return bCopyComponentUuid;
+  }
+
+  @CheckForNull
+  public String getBDescription() {
+    return bDescription;
+  }
+
+  public boolean isBEnabled() {
+    return bEnabled;
+  }
+
+  @CheckForNull
+  public String getBLanguage() {
+    return bLanguage;
+  }
+
+  @CheckForNull
+  public String getBLongName() {
+    return bLongName;
+  }
+
+  @CheckForNull
+  public String getBModuleUuid() {
+    return bModuleUuid;
+  }
+
+  @CheckForNull
+  public String getBModuleUuidPath() {
+    return bModuleUuidPath;
+  }
+
+  @CheckForNull
+  public String getBName() {
+    return bName;
+  }
+
+  @CheckForNull
+  public String getBPath() {
+    return bPath;
+  }
+
+  @CheckForNull
+  public String getBQualifier() {
+    return bQualifier;
+  }
+
+  public ComponentUpdateDto setBChanged(boolean b) {
+    this.bChanged = b;
+    return this;
+  }
+
+  public ComponentUpdateDto setBCopyComponentUuid(@Nullable String s) {
+    this.bCopyComponentUuid = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBEnabled(boolean b) {
+    this.bEnabled = b;
+    return this;
+  }
+
+  public ComponentUpdateDto setBName(@Nullable String s) {
+    this.bName = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBLongName(@Nullable String s) {
+    this.bLongName = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBDescription(@Nullable String s) {
+    this.bDescription = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBModuleUuid(@Nullable String s) {
+    this.bModuleUuid = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBModuleUuidPath(@Nullable String s) {
+    this.bModuleUuidPath = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBPath(@Nullable String s) {
+    this.bPath = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBLanguage(@Nullable String s) {
+    this.bLanguage = s;
+    return this;
+  }
+
+  public ComponentUpdateDto setBQualifier(@Nullable String s) {
+    this.bQualifier = s;
+    return this;
+  }
+
+  /**
+   * Copy the A-fields to B-fields. The field bChanged is kept to false.
+   */
+  public static ComponentUpdateDto copyFrom(ComponentDto from) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document the parameter(s): from [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this method return value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)